versions/development/SPEC.md

```diff
@@ -3091,6 +3092,20 @@ Map[String,Int] xmap = as_map(x) # {"a": 1, "b": 2, "c": 3}
 Map[String,Pair[File,File]] ymap = as_map(y) # {"a": ("a.bam", "a.bai"), "b": ("b.bam", "b.bai")}
 ```
 
+## Array[X] keys(Map[X,Y])
+
+Given an Map ,the `keys` function returns a Array of the keys of map. The order of the keys in the resulting Array is the same as the order of the Pairs in the Map.
```

Comment:
`Map ,the` should be `Map, the`.

I also feel this first sentence could read a bit awkwardly, maybe something like this: "Given a Map, the `keys` function returns an Array consisting of the keys in the Map."

Reply:
:+1:

Reply:
I think some engine implementations (maybe most) used OrdererdDicts I am not sure that it is always going to be the case, ie if an implementation is using a hash Dict.